File: 0137-single-number-ii/0137-single-number-ii.java
class Solution {
    public int singleNumber(int[] nums) {
        int ans = 0;
        for(int i = 0; i<32; i++) {
            int cnt = 0;
            for(var j : nums) cnt += (j>>i)&1;
            ans += (cnt%3) << i;
        }
        return ans;
    }
}

File: 0137-single-number-ii/0137-single-number-ii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = 0
        for i in range(32):
            cnt = 0
            for j in nums:
                if j < 0:
                    j = j & (2**32 - 1)
                cnt += (j >> i) & 1
            ans |= (cnt % 3) << i
        if ans >= 2**31:
            ans -= 2**32
        return ans
